### DISCUSSION TITLE
MJPC on a physical robot

### DISCUSSION THREAD

**Discussion by cygnus77** [google-deepmind/mujoco#2220] [Asking for Help]
### Intro

Hi! I'm using MJPC drive a robotic crane!

### My setup

Mujoco 3.1.4 MJPC

### My question

I'm using MJPC to drive a robotic crane to a target position. It is a simple model with 3 joints for controlling motion and a free joint on the object being transported.
I'm syncing real-time state of the physical crane with MuJoCo's model, running predictive sampling rollouts that generate control values, which are then fed to the crane over ROS. This loop runs at 100Hz.
Qpos, qvel from sensor readings are set in mjData along with ctrl.

Since mjData::act can't be sensed, what should it be set to?

I'm looking for advice/guidance on the proper way to import state from a robotic system into MuJoCo and how actuator states are handled.

Thanks in advance!

### Confirmations

- [X] I searched the [latest documentation](https://mujoco.readthedocs.io/en/latest/overview.html) thoroughly before posting.
- [X] I searched previous [Issues](https://github.com/google-deepmind/mujoco/issues) and [Discussions](https://github.com/google-deepmind/mujoco/discussions), I am certain this has not been raised before.

**Comment by yuvaltassa** [comment]
This question belongs in https://github.com/google-deepmind/mujoco_mpc

What do you mean "can't be sensed"? Of course it can be sensed. This is the internal state of your actuator, the controller should be reporting it. There should be a ROS message that reports this value.
